### PR TITLE
docs(ecs-infra): ✨ EC2/External tutorial using Daemon Service

### PR DESCRIPTION
**Pull request by prashant-shahi**
Signed-off-by: Prashant Shahi <prashant@signoz.io>

### CODE REVIEW

File: docs/shared/ecs-infra/daemon-service-template-ec2.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
+*daemon-template.yaml*
```
Comment (nityanandagohain) [inline docs/shared/ecs-infra/daemon-service-template-ec2.md]: Add as step 1 and mention we are going to use this file in step 3

File: docs/shared/ecs-infra/daemon-service-config-intro.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+### Step 2: Create SigNoz OtelCollector Config
```
Comment (nityanandagohain) [inline docs/shared/ecs-infra/daemon-service-config-intro.md:1]: Add image

**Review by nityanandagohain — CHANGES_REQUESTED** [review]

File: docs/shared/ecs-infra/daemon-service-set-up-verify.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+### Step 3: Configure Daemon Service
+
+To configure the daemon service, you need to run the following command with the proper values for the parameters.
```
Comment (nityanandagohain) [inline docs/shared/ecs-infra/daemon-service-set-up-verify.md]: Ask user to replace the values of clyster name and ecs region explicitly.

File: docs/shared/ecs-infra/daemon-service-set-up-verify.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+```bash
+export CLUSTER_NAME=<your-ecs-cluster-name>
+export REGION=<your-ecs-region>
+export COMMAND=--config=env:SIGNOZ_CONFIG_CONTENT
+export SIGNOZ_CONFIG=/ecs/signoz/otelcol-daemon.yaml
+
+aws cloudformation create-stack --stack-name AOCECS-daemon-${CLUSTER_NAME}-${REGION} \
```
Comment (nityanandagohain) [inline docs/shared/ecs-infra/daemon-service-set-up-verify.md:20]: seperate into two blocks as one needs replacement and the other doesn't.

File: docs/tutorial/ecs-infra.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+## Setting up Daemon Service
+
+<DSTemplateIntro name="Cloud" />
+
+<Tabs groupId="launch-type">
+<TabItem value="ec2" label="EC2" default>
+<DSTemplateEC2 />
+</TabItem>
+<TabItem value="external" label="External">
+<DSTemplateExternal />
```
Comment (nityanandagohain) [inline docs/tutorial/ecs-infra.md]: Add that fargate is not support yet

File: docs/tutorial/ecs-infra.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+### Prerequisites
+
+- An ECS cluster running with at least one task definition
+- SigNoz Cloud account - [Sign up for SigNoz Cloud](https://signoz.io/teams/)
+- Access Token and ingest URL, provided for your SigNoz Cloud account
+
+## Setting up Daemon Service
+
+<DSTemplateIntro name="Cloud" />
+
```
Comment (nityanandagohain) [inline docs/tutorial/ecs-infra.md]: select the type of ecs cluster

**Comment by nityanandagohain** [comment]
Add docs for cleanup as well.

File: docs/shared/ecs-infra/daemon-service-config-cloud.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+- Depending on the choice of your region for SigNoz cloud, the OTLP endpoint
+  will vary according to this table.
+
+  | Region	| Endpoint |
+  | --- | --- |
+  | US | ingest.us.signoz.cloud:443 |
+  | IN | ingest.in.signoz.cloud:443 |
+  | EU | ingest.eu.signoz.cloud:443 |
+- Include SigNoz ingestion key in place of `<signoz-ingestion-token>`
```
Comment (Calm-Rock) [inline docs/shared/ecs-infra/daemon-service-config-cloud.md]:
```suggestion
- Include SigNoz ingestion key in place of `<signoz-ingestion-token>`
```
We can remove this bullet point as we are mentioning about `SIGNOZ_INGESTION_KEY` in the next bullet point.

File: docs/shared/ecs-infra/daemon-service-config-cloud.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+- Replace `SIGNOZ_INGESTION_KEY` with the ingestion token of your SigNoz Cloud account
```
Comment (Calm-Rock) [inline docs/shared/ecs-infra/daemon-service-config-cloud.md]:
```suggestion
- Replace `SIGNOZ_INGESTION_KEY` with the ingestion key of your SigNoz Cloud account
```

File: docs/shared/ecs-infra/daemon-service-config-cloud.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+- After successful set up, feel free to remove `logging` exporter if it gets too noisy.
```
Comment (Calm-Rock) [inline docs/shared/ecs-infra/daemon-service-config-cloud.md]: Below this line we should mention

You can find the `SigNoz Ingestion Key` and `region` details in the invitation email or the Settings Tab of your SigNoz cloud UI.

File: docs/shared/ecs-infra/daemon-service-set-up-verify.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    --template-body file://daemon-template.yaml \
+    --parameters ParameterKey=ClusterName,ParameterValue=${CLUSTER_NAME} \
+    ParameterKey=CreateIAMRoles,ParameterValue=True \
+		ParameterKey=command,ParameterValue=${COMMAND} \
+		ParameterKey=SigNozConfig,ParameterValue=${SIGNOZ_CONFIG} \
+    --capabilities CAPABILITY_NAMED_IAM \
+    --region ${REGION}
+```
```
Comment (Calm-Rock) [inline docs/shared/ecs-infra/daemon-service-set-up-verify.md:28]: Please mention about what these variables represent
- `REGION`
- `CLUSTER_NAME`
- `COMMAND`
- `SIGNOZ_CONFIG_CONTENT`
- `SIGNOZ_CONFIG`

File: docs/shared/ecs-infra/daemon-service-set-up-verify.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+### Step 4: Verify Daemon Service
+
+To verify that the daemon service is running, you can run the following command.
+
+```bash
+aws ecs list-tasks --cluster ${CLUSTER_NAME} --region ${REGION}
+```
```
Comment (Calm-Rock) [inline docs/shared/ecs-infra/daemon-service-set-up-verify.md:36]: Here as well
What does `REGION` represent ? Is it the AWS region or the SigNoz Cloud Region ?
Create a bullet point explaining this.

Also include a bullet point for what `CLUSTER_NAME` Is here as well.

File: docs/shared/ecs-infra/daemon-service-send-data-intro.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+SigNoz {props.name} using Daemon Service we created in the previous section.
+</div>
+<br/>
+
+### Step 1: Add OpenTelemetry Instrumentation to your Application
+
+To add OpenTelemetry instrumentation to your application, you can follow the
+docs [here](https://signoz.io/docs/instrumentation/).
+
+This step can also include adding the OpenTelemetry SDK as well as the
```
Comment (Calm-Rock) [inline docs/shared/ecs-infra/daemon-service-send-data-intro.md]: Can we add this as `:::note  :::` admonition and also hyperlink `OpenTelemetry SDK` to a relevant link for instrumentation using OpenTelemetry SDK ?

File: docs/shared/ecs-infra/daemon-service-send-data-ec2.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
+**Method 1: Updating entrypoint in task definition**
+
+```json
+{
+      ...,
+      "entryPoint": [
+        "sh",
+        "-c",
+        "export OTEL_EXPORTER_OTLP_ENDPOINT=\"http://$(curl http://169.254.169.254/latest/meta-data/local-ipv4):4317\"; <Application Startup Commands>"
```
Comment (Calm-Rock) [inline docs/shared/ecs-infra/daemon-service-send-data-ec2.md:15]: What is `<Application Startup Commands>`?
Can you add a bullet point explaining it?

File: docs/shared/ecs-infra/daemon-service-send-data-end.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+### Step 4: Rebuild and Deploy Application Container
+
+After following previous step, you need to rebuild the application container and deploy it to ECS cluster.
+
+### Step 5: Verify Data in SigNoz
+
+<div>
+You will need to generate some traffic to your application. After which you can go to the
+SigNoz {props.name} page and you should see your application in the services list.
```
Comment (Calm-Rock) [inline docs/shared/ecs-infra/daemon-service-send-data-end.md]: Can you mention which section of SigNoz Cloud will the user see the data ?
SigNoz Cloud logs/Metrics/Services whichever is relevant.

File: docs/shared/ecs-infra/daemon-service-send-data-external.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+    
+- If network mode is not **Bridge**, and GCP instance:
+    
+    ```json
+    {
+          ...,
+          "entryPoint": [
+            "sh",
+            "-c",
+    				'export OTEL_EXPORTER_OTLP_ENDPOINT="http://$(curl "http://169.254.169.254/computeMetadata/v1/instance/network-interfaces/0/ip?recursive=true&alt=text" -H "Metadata-Flavor: Google"):4317"; <Application Startup Commands>'
```
Comment (Calm-Rock) [inline docs/shared/ecs-infra/daemon-service-send-data-external.md:78]: What is `<Application Startup Commands>`?
Can you add a bullet point explaining it everywhere it is mentioned?

File: docs/shared/ecs-infra/daemon-service-template-ec2.md [hunk 16]
```diff
@@ -0,0 +1,194 @@
...
+          - ECSExecutionRole
+          - Arn
+        - !If
+          - DefaultExecutionRole
+          - !Sub 'arn:aws:iam::${AWS::AccountId}:role/OtelECSExecutionRole'
+          - !Ref ExecutionRoleArn
+      NetworkMode: host
+      ContainerDefinitions:
+        - Name: signoz-collector
+          Image: 'signoz/signoz-otel-collector:0.79.12'
```
Comment (nityanandagohain) [inline docs/shared/ecs-infra/daemon-service-template-ec2.md]: change it to 0.79.7

File: docs/shared/ecs-infra/daemon-service-set-up-verify.md (hunk 10 above)
Comment (prashant-shahi) [reply docs/shared/ecs-infra/daemon-service-set-up-verify.md:28]: Only `REGION` and `CLUSTER_NAME` are needed.

Adding them in an info block

File: docs/shared/ecs-infra/daemon-service-set-up-verify.md (hunk 11 above)
Comment (prashant-shahi) [reply docs/shared/ecs-infra/daemon-service-set-up-verify.md:36]: I think one info block in the beginning should suffice for this.

If required, we can come back and update this.

**Pull request merged**